src/plugins/intel_cpu/src/transformations/transformation_pipeline.cpp

```diff
@@ -728,13 +734,29 @@ void Transformations::MainSnippets(void) {
                                                         [&](const ov::Input<const ov::Node>& in) {
                                                             return rank_is_too_large(in.get_tensor());
                                                         });
+                if (bad_input_rank)
+                    return true;
                 const auto& outputs = n->outputs();
                 const bool bad_output_rank = std::any_of(outputs.begin(), outputs.end(),
                                                         [&](const ov::Output<const ov::Node>& out) {
                                                             return rank_is_too_large(out.get_tensor());
                                                         });
-                return has_only_const_inputs || bad_input_rank || bad_output_rank || is_unsupported_swish ||
-                    is_disabled_tokenization;
+                if (bad_output_rank)
+                    return true;
+                const bool has_dynamic_input = std::any_of(inputs.begin(), inputs.end(),
+                                                        [&](const ov::Input<const ov::Node>& in) {
+                                                            return !in.get_tensor().get_partial_shape().is_static();
```

Comment:
We already have the check for dynamic shapes 🙂 
```suggestion
                                                            return in.get_partial_shape().is_dynamic();
```

BTW The `ov::Node` has interface `is_dynamic()` which check for dynamic shapes of inputs. Since Snippets don't support nodes with internal dynamism we don't expect to get node with input static shapes but dynamic output shapes (I understand that probably it will be supported in the future but you have already insert early `returns` to make callback quicker). What do you think can we reuse here just `node->is_dynamic()`?

Also there may be sense to reorder checks here. For example, insert `checks for dynamism` earlier than checks for `HSwish`. Maybe not in this PR...